### PR TITLE
OSDOCS-8729: Reviewed AWS Secrets Manager on ROSA migration

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
`enterprise-4.14+`

Issue:
[OSDOCS-8729](https://issues.redhat.com/browse/OSDOCS-8729)

Link to docs preview:
- [Using AWS Secrets Manager CSI on ROSA with STS](https://file.rdu.redhat.com/eponvell/OSDOCS-8729_AWS-Secrets-Manager-MOBB/cloud_experts_tutorials/cloud-experts-aws-secret-manager.html)

QE review:
- [x] QE has approved this change.

Additional information:
Reviewed "Using AWS Secrets Manager CSI on ROSA with STS" for style and clarity. This PR requires QE approval.

### CODE REVIEW

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 1]
```diff
@@ -17,38 +17,49 @@ toc::[]
...
+=== Additional environment requirements
 
-. Validate that your cluster has STS:
+. Log in to your ROSA cluster by running the following comnmand:
++[source,terminal]
+----
+$ oc login --token=<your-token> --server=<your-server-id>
+----
++
+You can find your log in token by accessing your cluster in {cluster-manager-url-pull}.
```
Comment (xingxingxia) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:43]: From the preview link, I see below effect:
Screenshot from 2024-01-08 20-47-53

```
1. Log in ... comnmand: +[source,terminal]
+ You can find your log ...
1. Validate that your cluster has STS ...
```
That is, it has problems:

1. `+[source,terminal]` is shown bad format.
2. Typo `comnmand`
3. `+ You can find your log` has bad format `+`
4. Two items of same starting "1." order.

@EricPonvelle  Suggest document peers review and find such format issues and fix them first before requesting QE review :)

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 2]
```diff
@@ -17,38 +17,50 @@ toc::[]
...
 [id="cloud-experts-aws-secret-manager-preparing-environment"]
-== Preparing Environment
+=== Additional environment requirements
 
-. Validate that your cluster has STS:
+. Log in to your ROSA cluster by running the following command:
++
+[source,terminal]
+----
+$ oc login --token=<your-token> --server=<your-server-id>
```
Comment (xingxingxia) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:41]:
```suggestion
$ oc login --token=<your-token> --server=<your-server-url>
```
Sorry did not notice this in last review. It is not "id", it is env api server url which looks like https://api.xxxxxxx:6443 .

**Review by xingxingxia — COMMENTED** [review]
This PR LGTM from my side now except a minor wording. No other issue from my side.

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 3]
```diff
@@ -17,38 +17,50 @@ toc::[]
...
 
-. Validate that your cluster has STS:
+. Log in to your ROSA cluster by running the following command:
++
+[source,terminal]
+----
+$ oc login --token=<your-token> --server=<your-server-url>
+----
++
+You can find your log in token by accessing your cluster in {cluster-manager-url-pull}.
```
Comment (bburt-rh) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:44]:
```suggestion
You can find your login token by accessing your cluster in {cluster-manager-url-pull}.
```
Per [the ISG word usage guidelines](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#l), it should be `login` when used as an adjective.

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 4]
```diff
@@ -17,38 +17,50 @@ toc::[]
...
 +
-You should see something like the following, if not you should not proceed, instead look to the link:https://docs.openshift.com/rosa/rosa_getting_started_sts/rosa_creating_a_cluster_with_sts/rosa-sts-creating-a-cluster-quickly.html[Red Hat documentation on creating an STS cluster].
+You should see the following:
 +
 [source,terminal]
 ----
 "https://xxxxx.cloudfront.net/xxxxx"
 ----
++
+If your output is different, you should not proceed. See xref:../rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc#rosa-sts-creating-a-cluster-quickly[Red Hat documentation on creating an STS cluster] before continuing this process.
```
Comment (bburt-rh) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:61]:
```suggestion
If your output is different, do not proceed. See xref:../rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc#rosa-sts-creating-a-cluster-quickly[Red Hat documentation on creating an STS cluster] before continuing this process.
```
- Per the ISG, avoid "should" (see my previous comment). It's OK to be prescriptive and authoritative here, I think.

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 5]
```diff
@@ -17,38 +17,50 @@ toc::[]
...
 
-. Set SecurityContextConstraints to allow the CSI driver to run:
+. Set `SecurityContextConstraints` to allow the CSI driver to run:
```
Comment (bburt-rh) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:63]:
```suggestion
. Set the `SecurityContextConstraints` [object?] to allow the CSI driver to run by running the following command:
```
- The literal needs a referent. Is it an object? I'm not sure.
- Add `by running the following command` per [the OCP docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command).

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 6]
```diff
@@ -17,38 +17,50 @@ toc::[]
...
 +
 [source,terminal]
 ----
 $ oc get authentication.config.openshift.io cluster -o json \
   | jq .spec.serviceAccountIssuer
 ----
 +
-You should see something like the following, if not you should not proceed, instead look to the link:https://docs.openshift.com/rosa/rosa_getting_started_sts/rosa_creating_a_cluster_with_sts/rosa-sts-creating-a-cluster-quickly.html[Red Hat documentation on creating an STS cluster].
+You should see the following:
 +
```
Comment (bburt-rh) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:55]:
```suggestion
.Example output
```
- Per [the IBM word usage guidelines](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s), avoid "should" and rewrite in a more direct way. But in any case, I suggest that you delete this sentence entirely and replace it with an `.Example output` heading.
- Per [the IBM word usage guidelines](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#f), for introductory phrases, always use "following" as an adjective followed by a noun.

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 7]
```diff
@@ -59,7 +71,7 @@ $ oc adm policy add-scc-to-user privileged \
     system:serviceaccount:csi-secrets-store:csi-secrets-store-provider-aws
 ----
 
-. Create some environment variables to refer to later:
+. Create some environment variables to use later in this process:
```
Comment (bburt-rh) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:74]:
```suggestion
. Create environment variables to use later in this process:
```
- `some` is unnecessary, imo

File: cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc [hunk 8]
```diff
@@ -248,7 +260,7 @@ spec:
 EOF
 ----
 
-. Create a Deployment using our secret:
+. Create a Deployment by using our secret:
```
Comment (bburt-rh) [inline cloud_experts_tutorials/cloud-experts-aws-secret-manager.adoc:263]:
```suggestion
. Create a Deployment by using our secret in the following command:
```

**Review by bburt-rh — APPROVED** [review]
@EricPonvelle - Thanks for this PR! I've suggested some changes to this PR per the ISG and OCP docs guidelines.

I also took a quick look at the unchanged procedures in this assembly, and I noticed that very few of the commands are following the OCP standard for introducing a CLI command in a step: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command. Not sure if these changes would be in scope for this PR, but since you are already in the file, maybe now's the time to make these corrections? WDYT.

Thanks again.

**Pull request merged**